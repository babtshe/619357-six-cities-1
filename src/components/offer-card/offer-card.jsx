import React, {useCallback} from 'react';
import {propTypes} from './offer-card.props';
import {Link} from 'react-router-dom';
import BookmarkButton from '../bookmark-button/bookmark-button';
import withToggleBookmark from '../../hocs/with-toggle-bookmark/with-toggle-bookmark';

const ToggleBookmarkWrapped = withToggleBookmark(BookmarkButton);

const OfferCard = (props) => {
  const {
    card,
    onClick,
    classPrefix,
  } = props;
  const handleMouseClick = useCallback((evt) => {
    evt.preventDefault();
    if (typeof onClick === `function`) {
      onClick(card);
    }
  }, [card]);

  return (
    <article className={`${classPrefix}__place-card place-card`}>
      {!!card.premium && (
        <div className="place-card__mark">
          <span>Premium</span>
        </div>
      )}
      <div className={`${classPrefix}__image-wrapper place-card__image-wrapper`}>
        <a href="#" onClick={handleMouseClick}>
          <img className="place-card__image" src={card.image} width="260" height="200"
            alt="Place image" />
        </a>
      </div>
      <div className={`${classPrefix}__card-info place-card__info`}>
        <div className="place-card__price-wrapper">
          <div className="place-card__price">
            <b className="place-card__price-value">&euro;{card.price}</b>
            <span className="place-card__price-text">&#47;&nbsp;night</span>
          </div>
          <ToggleBookmarkWrapped
            id={card.id}
            bookmarked={card.bookmarked}
            isSmall={true}
          />
        </div>
        <div className="place-card__rating rating">
          <div className="place-card__stars rating__stars">
            <span style={{width: `${card.rating}%`}}></span>
            <span className="visually-hidden">Rating</span>
          </div>
        </div>
        <h2 className="place-card__name">
          <Link to={`/offer/${card.id}`}>{card.name}</Link>
        </h2>
        <p className="place-card__type">{card.type}</p>
      </div>
    </article>
  );
};

OfferCard.propTypes = propTypes;

export {OfferCard};
