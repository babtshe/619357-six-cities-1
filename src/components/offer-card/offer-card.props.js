import PropTypes from 'prop-types';
import {coordinates} from '../../custom-prop-types/coordinates.prop';

export const cardTypes = {
  name: PropTypes.string.isRequired,
  type: PropTypes.oneOf([`apartment`, `room`, `house`, `hotel`]).isRequired,
  price: PropTypes.number.isRequired,
  rating: PropTypes.number.isRequired,
  image: PropTypes.string.isRequired,
  bookmarked: PropTypes.bool,
  premium: PropTypes.bool,
  location: coordinates.isRequired,
};

export const propTypes = {
  card: PropTypes.shape(cardTypes),
  onClick: PropTypes.func.isRequired,
  onMouseEnter: PropTypes.func.isRequired,
  onMouseLeave: PropTypes.func.isRequired,
};
