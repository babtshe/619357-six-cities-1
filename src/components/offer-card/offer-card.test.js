import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import {OfferCard} from './offer-card';

it(`OfferCard should render correctly`, ()=>{
  const mockCard = {
    name: `test1`,
    type: `apartment`,
    price: 1,
    rating: 1,
    image: `test2`,
    premium: true,
    id: 1,
    location: [52.3909553943508, 4.85309666406198],
  };
  const renderer = new ShallowRenderer();
  const tree = renderer
  .render(
      <OfferCard
        card = {mockCard}
        onClick = {()=>{}}
        onMouseEnter = {()=>{}}
        onMouseLeave = {()=>{}}
      />);

  expect(tree).toMatchSnapshot();
});
