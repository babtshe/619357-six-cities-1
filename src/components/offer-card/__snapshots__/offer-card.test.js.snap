// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferCard should render correctly 1`] = `
<article
  className="undefined__place-card place-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="place-card__mark"
  >
    <span>
      Premium
    </span>
  </div>
  <div
    className="undefined__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
      onClick={[Function]}
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="test2"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          1
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <Memo(ConnectFunction)
        id={1}
        isSmall={true}
      />
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "1%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <Link
        to="/offer/1"
      >
        test1
      </Link>
    </h2>
    <p
      className="place-card__type"
    >
      apartment
    </p>
  </div>
</article>
`;
